javascript-stringify.js

```diff
@@ -149,19 +159,59 @@
   function stringifyObject (object, indent, next) {
     // Iterate over object keys and concat string together.
     var values = Object.keys(object).reduce(function (values, key) {
-      var value = next(object[key], key);
+      var value;
+      var addKey = true;
+
+      // Handle functions specially to detect method notation.
+      if (typeof object[key] === 'function') {
+        var fn = object[key];
+        var fnString = fn.toString();
+        var prefix = isGeneratorFunction(fn) ? '*' : '';
+
+        // Was this function defined with method notation?
+        if (fn.name === key && fnString.startsWith(prefix + key + '(')) {
```

Comment:
Just a heads up but `startsWith` is ES2015. Probably not an issue and will look at updating to ES2015 soon.